client/client.cpp

```diff
@@ -2,105 +2,139 @@
+
+    void runThreaded() {
+        std::vector<std::thread> threads;
+        for (int i = 0; i < num_clients; i++) {
+            threads.push_back(std::thread(&ClientHandler::run, this));
+        }
+
+        // Wait for all threads to finish
+        for (std::thread &thread : threads)
+            thread.join();
+
+        getAveLatency();
+    }
+
+    void run() {
+        auto socket = std::make_shared<TSocket>(HOST_IP, HOST_PORT);
+        auto transport = std::make_shared<TBufferedTransport>(socket);
+        auto protocol = std::make_shared<TBinaryProtocol>(transport);
+        RPCClient client(protocol);
+
+        transport->open();
+
+        std::string val;
+        // If seed data exists, run the client with data
+        if (seed_data.is_open()) {
+            std::string line;
+            while (readFile(seed_data, line)) {
+                Operation op = getSeedOperation(line);
+                auto start = high_resolution_clock::now();
+                client.access(val, op);
+                auto end = high_resolution_clock::now();
+                latencies.push_back(
+                    duration_cast<microseconds>(end - start).count());
+            }
+        }
+        // If seed data does not exist, run client on random values
+        else {
+            for (int i = 0; i < 1000; ++i) {
+                Operation op = genRandOperation(p_get);
+                auto start = high_resolution_clock::now();
+                client.access(val, op);
+                auto end = high_resolution_clock::now();
+                latencies.push_back(
+                    duration_cast<microseconds>(end - start).count());
+            }
+        }
+
+        transport->close();
+    }
+
+    void getAveLatency() {
+        std::cout << "[Client]: Data access complete, average latency: "
+                  << std::accumulate(latencies.begin(), latencies.end(), 0.0) /
+                         latencies.size()
+                  << " microseconds" << std::endl;
+    }
 };
 
-
 int main(int argc, char *argv[]) {
-  auto start = high_resolution_clock::now();
-
-	// If user runs client with path to seed data, init ClientHandler with seed
-	try {
-		ClientHandler client;
-		if (argc >= 2){
-			std::string seed_data_path = argv[1];
-			client = ClientHandler(seed_data_path);
-		}
-		client.run_threaded();
-		client.getAveLatency();
-	} catch (std::invalid_argument& err) {
-		std::cerr << "ERROR: " << err.what() << std::endl;
-	} catch (TException& err) {
-		std::cerr << "ERROR: " << err.what() << std::endl;
-  }
-
-	auto end = high_resolution_clock::now();
-  std::cout << "[main]: Entire program finished in " << duration_cast<microseconds>(end - start).count() << " microseconds" << std::endl;
+    try {
+        ClientHandler client(argc, argv);
+
+        auto start = high_resolution_clock::now();
+        client.start();
+        auto end = high_resolution_clock::now();
+
+        std::cout << "[main]: Entire program finished in "
```

Comment:
In a separate PR we should also change this stuff to use the logger :)

Reply:
Yep I added a ticket for us to address this